CONTRIBUTING.md

```diff
@@ -33,22 +53,53 @@ Here are the steps for contributing to ECS.
 * Request feedback and permission to merge your changes.
   - Create a [Pull Request](https://help.github.com/articles/creating-a-pull-request/) against the ECS repo.
   - (Look for the `Compare & pull request` button on your branch in github.com.)
-* Next steps
   - Wait for reviews on your PR.
   - Incorporate review comments and push updates if needed.
 * Thank you for your contribution!
 
 **Important:** Be sure to push changes only to your own fork. Changes must be approved before they are merged into the main repository.
 
-### Other ways to contribute
+## Git and Github Guidelines
+
+### Forking
+
+We follow the [Github forking model](https://help.github.com/articles/fork-a-repo/) for collaboration in the ECS repo. Typically contributors will add a remote repository called `upstream` to point to the Elastic ECS repo to add latest changes from the repo to their fork.
+
+### Commits and Merging
+
+* When submitting a PR for review, please perform and interactive rebase to clean up the commit history. A logical commit history is easier for reviewers to follow.
+* Use meaningful and helpful commit messages on your changes and an explanation of _why_ you made those changes.
+* When merging, maintainers will squash your commits into a single commit.
+
+### Pull Requests
+
+Please follow these guidelines when submitting PRs:
+
+* Include an explanation of your changes in the PR description.
+* Links to relevant issues, external resources, or related PRs are helpful and useful.
+* Update any tests or add new tests where appropriate.
+
+### Issues
+
+Please follow these guidlines when submitting Issues:
 
-You can contribute even if you are not an experienced Git user. You'll need a github.com account.
 * Go to the ECS repo: https://github.com/elastic/ecs
 * Click `Issues` in the nav bar under the repo name.
 * Click `New issue`. Provide as many details as possible to help reviewers and other contributors understand your proposal.
 * Add your text, and click `Submit new issue`.
 
-## Fields.yml
+
+## Documentation
+
+ECS documentation is written in [asciidoc](http://asciidoc.org/) format in the `docs/` directory.
+
+To build the docs and open them in your browser:
+
+```bash
+make docs
+```
+
+## Schema Files
 
 The `fields.yml` files describe the Elastic Common Schema in a structured way. We can use these files to generate an Elasticsearch index template, a Kibana index pattern, or documentation output.
```

Comment:
Whoa, I think this is from before my time 😲 I think `fields.yml` actually refers to the files in `schemas/*.yml`

Can you reword this a bit? It no longer needs to talk about possibilities, we do all of this now (except Kibana index patterns, and we won't get into that).

Also, this is a great time to tell reinforce that `schemas/*.yml` is where field additions/modifications are made, but if users are looking to consume ECS to generate something else, they should instead head to `generated/ecs/*.yml`

Reply:
Let me know what you think of the updated wording here.

I can see moving the example `agent` snippet into the `schemas/README.md` and linking to it from `CONTRIBUTING.md`. The idea being to have `schemas/README.md` be the single place in the repo with the schema definition, examples, etc.

Reply:
Yes for now let's completely remove the section about `agent`, I agree.

However please don't modify `schemas/README.md` for now, as I've dramatically improved it in the `nest-as-rampage` branch. You can check it out [here](https://github.com/webmat/ecs/blob/nest-as-rampage/schemas/README.md)